File: Source/server/rabbit-game/src/SharedModel/Messages/RoomResponseMsg.cs

namespace RabbitGameServer.SharedModel.Messages
{
	public class RoomResponseMsg : Message
	{
		public RoomResponseType responseType;

		public List<PlayerData> players;

		public RoomResponseMsg(DateTime timestamp,
			RoomResponseType type,
			string player,
			string roomName,
			List<PlayerData> players)
			: base(MessageType.RoomResponse, timestamp, player, roomName)
		{
			this.responseType = type;
			this.players = players;
		}

		public static RoomResponseMsg unknownFail()
		{
			return new RoomResponseMsg(DateTime.Now, RoomResponseType.UnknownFail, null, null, null);
		}

		public static RoomResponseMsg success(string requester,
				string roomName,
				List<PlayerData> players)
		{

			return new RoomResponseMsg(DateTime.Now, RoomResponseType.Success, requester, roomName, players);
		}

	}
}

File: Source/root/src/main/java/root/communication/messages/RoomResponseMsg.java
package root.communication.messages;

import java.util.Date;
import java.util.List;

import root.communication.PlayerDescription;

public class RoomResponseMsg extends Message {

	public RoomResponseType responseType;

	// initialize even in createRoom request and 
	// add only one element containing description of request sender
	public List<PlayerDescription> players;

	public RoomResponseMsg(Date timestamp,RoomResponseType responseType, List<PlayerDescription> players) {
		super(MessageType.RoomResponse, timestamp);

		this.responseType = responseType;
		this.players = players;
	}

}
